tests/src/Unit/AttributesExplicitTest.php

```diff
@@ -110,6 +110,48 @@ public function testUnusualAttributeValueStringsEnclosed() {
     }
   }
 
+  /**
+   * Tests unusual non-string attribute values.
+   */
+  public function testUnusualAttributeValues() {
+
+    foreach ([
+      [TRUE, TRUE],
+      [FALSE, TRUE],
+      [[TRUE], '1'],
+      [[TRUE, TRUE], '1'],
+      [[FALSE, FALSE], ''],
+      [0, '0'],
+      [3, '3'],
+      [[1, 2, 3], '1 2 3'],
+      [[1, [2, [3]]], '1 2 3'],
+      [M_PI, (string) M_PI],
+      [[M_PI], (string) M_PI],
+      [[M_PI, [M_PI + 1]], M_PI . ' ' . (M_PI + 1)],
+      [['a', 'b', 'c'], 'a b c'],
+      [['a ', ' b ', ' c ', ' d ', ' e'], 'a b c d e'],
+      [['a', ['b', ['c']]], 'a b c'],
+    ] as $value_and_expected) {
+
+      list($value, $expected_value) = $value_and_expected;
```

Comment:
@drupol I would like to have this list() inside the foreach statement, but grumphp gets confused when I do that.

```php
foreach ($array as list($a, $b)) {
  ..
}
```
https://3v4l.org/2fVDn